pkg/controller/controlplane/provider_actuator.go

```diff
@@ -0,0 +1,142 @@
+	configName string,
+	shootWebhooks []admissionregistrationv1beta1.MutatingWebhook,
+	webhookServerPort int,
+	logger logr.Logger,
+) controlplane.Actuator {
+	logger = logger.WithName("controlplane-actuator")
+
+	return &providerActuator{
+		genericActuator: genericactuator.NewActuator(
+			providerName,
+			secrets,
+			exposureSecrets,
+			configChart,
+			controlPlaneChart,
+			controlPlaneShootChart,
+			storageClassesChart,
+			controlPlaneExposureChart,
+			vp,
+			chartRendererFactory,
+			imageVector,
+			configName,
+			shootWebhooks,
+			webhookServerPort,
+			logger,
+		),
+	}
+}
+
+// InjectFunc enables injecting Kubernetes dependencies into actuator's dependencies.
+func (a *providerActuator) InjectFunc(f inject.Func) error {
+	injector, ok := a.genericActuator.(inject.Injector)
+	if ok {
+		return injector.InjectFunc(f)
+	}
+
+	return fmt.Errorf("Cotroller generic actuator does not implement Injector interface")
+}
+
+// InjectConfig injects the given config into the actuator.
+func (a *providerActuator) InjectConfig(config *rest.Config) error {
+
+	injector, ok := a.genericActuator.(inject.Config)
+	if ok {
+		return injector.InjectConfig(config)
+	}
+
+	return fmt.Errorf("Cotroller generic actuator does not implement Config interface")
+}
+
+// InjectClient injects the given client into the valuesProvider.
+func (a *providerActuator) InjectClient(client client.Client) error {
+	injector, ok := a.genericActuator.(inject.Client)
+	if ok {
+		a.client = client
+		return injector.InjectClient(client)
+	}
+
+	return fmt.Errorf("Cotroller generic actuator does not implement Config interface")
+}
+
+func (a *providerActuator) Reconcile(ctx context.Context, cp *extensionsv1alpha1.ControlPlane, cluster *extensionscontroller.Cluster) (bool, error) {
+	//Delete leaked resources
+	a.logger.Info("Cleanup leaked resources", "controlplane", kutil.ObjectName(cp))
+	if err := a.client.Delete(ctx, &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "cloud-controller-manager-monitoring-config", Namespace: cp.Namespace}}); err != nil && !errors.IsNotFound(err) {
+		return false, err
+	}
+
+	if err := a.client.Delete(ctx, &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: "cloud-controller-manager", Namespace: cp.Namespace}}); err != nil && !errors.IsNotFound(err) {
+		return false, err
+	}
+
+	return a.genericActuator.Reconcile(ctx, cp, cluster)
+}
+
+func (a *providerActuator) Delete(ctx context.Context, cp *extensionsv1alpha1.ControlPlane, cluster *extensionscontroller.Cluster) error {
+	return a.genericActuator.Delete(ctx, cp, cluster)
+}
+
+func (a *providerActuator) Restore(ctx context.Context, cp *extensionsv1alpha1.ControlPlane, cluster *extensionscontroller.Cluster) (bool, error) {
+	return a.genericActuator.Restore(ctx, cp, cluster)
+}
+
+func (a *providerActuator) Migrate(ctx context.Context, cp *extensionsv1alpha1.ControlPlane, cluster *extensionscontroller.Cluster) error {
+	return a.genericActuator.Migrate(ctx, cp, cluster)
+}
```

Comment:
With above suggestion (inline the generic actuator into the providerActuator struct) these lines could probably be deleted.